site/blog/tutorials/iasql-on-ec2.mdx

```diff
@@ -0,0 +1,284 @@
+
+##### Bash Startup Script
+
+Alright, we can now write our query to deploy IaSQL on EC2. Let's first write the query with EC2 startup script being a shell script, and then write the query with same functionality using [cloud-init](https://cloud-init.io/) directives.
+
+```sql
+INSERT INTO instance (ami, instance_type, subnet_id, tags, user_data, key_pair_name, region)
+SELECT 'resolve:ssm:/aws/service/canonical/ubuntu/server/22.04/stable/current/amd64/hvm/ebs-gp2/ami-id',
+       't2.small',
+       subnet.id,
+       '{"name":"iasql"}',
+       '#!/bin/bash
+     sudo apt-get upgrade -y
+     curl -fsSL https://get.docker.com -o get-docker.sh
+     sudo sh get-docker.sh
+     docker run -p 9876:9876 -p 5432:5432 --name iasql iasql/iasql',
+       'iasql',
+       'us-east-1'
+FROM subnet
+         INNER JOIN vpc ON vpc.id = subnet.vpc_id
+    AND vpc.is_default
+WHERE vpc.region = 'us-east-1'
+  AND subnet.availability_zone = 'us-east-1a'
+LIMIT 1;
+```
+
+There's just one last step needed. Remember the security group we created earlier? We never assigned our instance to it. So let's do it and let IaSQL do the rest:
+```sql
+INSERT INTO instance_security_groups (instance_id, security_group_id)
+SELECT (SELECT id
+        FROM instance
+        WHERE tags ->> 'name' = 'iasql'
+        LIMIT 1),
+       (SELECT id
+        FROM security_group
+        WHERE group_name = 'iasql_sg'
+        LIMIT 1);
+```
+The `instace_security_groups` table is a Many-to-Many join table that connects `instance`s to their `security_group`s. Now we need to wait for IaSQL to start syncing the state of the cloud with our local changes. This will normally happen every 2 minutes. There is a way to make the cloud apply/sync happen immediately using [IaSQL transactions](https://iasql.com/docs/transaction/), but for the purpose of this post we'll wait for the next cronjob tick to apply our changes.
+
+After waiting long enough for the server to start and for the startup script to finish, we can use instance's public IP to access our self-hosted IaSQL dashboard:
+
+```sql
+SELECT public_ip_address
+FROM instance_metadata
+WHERE id = (SELECT id
+            FROM instance
+            WHERE tags ->> 'name' = 'iasql'
+            LIMIT 1);
+```
+
+Now we can go to `http://<instance-ip>:9876/` URL, and we'll see the beautiful IaSQL dashboard deployed on our EC2 server.
+
+##### Cloud-init Directives
+Now let's use cloud-init directives to do the same thing. This part is not exactly related to IaSQL, or what we intended to do in this post. It's just an exploration for other choices we have with EC2 user data field.
```

Comment:
Since this section as you say is not directly related to the scope of this article can we explicitly mark it as optional or something like that?

Reply:
Is there a class for marking section as optional? Or something similiar?

Reply:
no idea!

Reply:
There is collapsible parts in `mdx`, but it's making the markdown really hard to edit.

Reply:
maybe just add an  something like:
```suggestion
##### Advanced: Cloud-init Directives
Now let's use cloud-init directives to do the same thing. This part is not exactly related to IaSQL, or what we intended to do in this post. It's just an exploration for other choices we have with EC2 user data field.
```

or 

```suggestion
##### Optional: Cloud-init Directives
Now let's use cloud-init directives to do the same thing. This part is not exactly related to IaSQL, or what we intended to do in this post. It's just an exploration for other choices we have with EC2 user data field.
```

Reply:
Used the second one. Thanks!